scripts/bert/utils.py

```diff
@@ -102,3 +103,40 @@ def read_tf_checkpoint(path):
         tensor = reader.get_tensor(key)
         tensors[key] = tensor
     return tensors
+
+
+def load_vocab(vocab_file):
+    """Loads a vocabulary file into a dictionary."""
+    vocab = collections.OrderedDict()
+    index = 0
+    with io.open(vocab_file, 'r') as reader:
+        while True:
+            token = reader.readline()
+            if not token:
+                break
+            token = token.strip()
+            vocab[token] = index
+            index += 1
+    return vocab
+
+
+def printable_text(text):
+    """Returns text encoded in a way suitable for print."""
+    # These functions want `str` for both Python2 and Python3, but in one case
+    # it's a Unicode string and in the other it's a byte string.
```

Comment:
Why do we have byte-string `str` in Py2 and not unicode-string `u`? Maybe the data reading code can be changed somewhere to read unicode both on Py2 and Py3. We should try to be consistent and use either byte on both Py2 and Py3, or unicode on both. Less chances for unintended bugs then.

In summary: Is this function really required? Or does the need for this function just point out that some other code does not handle the data reading correctly?

Reply:
@leezu This function is mainly designed for specific string generation in ` create_pretraining_data.py `.  It has not been considered before that it has been assumed that the user input data is already unicode encoded. If on the assumpaion as discussed earlier,  I might be able to remove the byte consideration in code.  @eric-haibin-lin